### DISCUSSION TITLE
Use Classification Extension properties for mosaic visualization

### DISCUSSION THREAD

**Discussion by hrodmn** [stac-utils/titiler-pgstac#145] [Ideas]
I don't know how widely used it is, but the [Classification STAC extension](https://github.com/stac-extensions/classification) can provide color hints and descriptions for classified rasters. Could we have titiler-pgstac look for metadata from this extension? In particular it seems like the [`color_hint`](https://pystac.readthedocs.io/en/stable/api/extensions/classification.html#pystac.extensions.classification.Classification.color_hint) parameter could be useful.

This might be useful for rendering classified rasters (e.g. land cover datasets) using the color hints defined by the data provider.

**Comment by vincentsarago** [comment]
👋 @hrodmn 

that's interesting, sadly it's hard to keep up with all the extensions! I usually wait for them to be official (the classification is at `Pilot` state. But we can totally think about it and see how to use it.

I realize there is another problem with the STAC endpoints, if the original dataset has colormap information we have no way to forward it to the `mosaic` 😓. This could be resolved at rio-tiler level.

between the virtual-asset, render, raster and classification there are too many option right now so it might be tricky to support `external colormap`
